### PR TITLE
CMS Integration

### LINKED ISSUE #229: Migrate About page to Gatsby/NetlifyCMS

### PR DESCRIPTION
**Pull request by BenRauzi**
Resolves #229 , #230, #231, #273.
Added CMS to About Us, Plugins, Developers pages.
Added Samuel Jenks + Ben Rauzi Contributor Cards.

This is an initial base for the CMS integration. In the next few days I will further modify it to make it easier to edit the developers and 'about us' pages in further PR's. (Each item will be a paragraph / one element, instead of the whole page's contents being within one page).

One issue is that I haven't got custom elements working with it (<Link>, <External>, etc) - noticed the same issue with the already created news items as they'd just been switched to <a> - therefore causing a reload and defeating part of the purpose of an SPA. 
I will look into a solution for this.
(Although that being said users should navigate using the navbar and there are very few internal links.)
___
NB - by making a pull request you confirm agreement to the [Contributors Agreement](https://github.com/openlawnz/openlawnz-web/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: content/about/primary-page-content.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
...
...ef=\"#infrastructure\">\r\n                    <img src=\"/assets/openlaw-infrastructure.png\" alt=\"Infrastructure\" title=\"Click to enlarge the image\" />\r\n                  </a>\r\n                </div>\r\n              </div>\r\n              <div aria-hidden=\"true\" id=\"infrastructure\" class=\"modal-window\">\r\n                <div>\r\n                  <a href=\"#close\" title=\"Close\" class=\"modal-close\">\r\n                    Close\r\n                  </a>\r\n                  <img src=\"/assets/openlaw-infrastructure.png\" alt=\"Infrastructure\" />\r\n                </div>\r\n              </div>\n         \n</div>\n\n <h2>Who we are</h2>\r\n            <p>\r\n              OpenLaw NZ is a registered charity. We’re operated by a fantastic group of volunteers, helping with code,\r\n              design, orchestration, data parsing, machine learning and devops.\r\n            </p>\r\n            <p>\r\n              Our volunteers receive real-world experience in react, node.js, python and a range of other technologies. If\r\n              you’d like to get involved, drop us a line <a href=\"mailto:enquiries@openlaw.nz\">here</a>.\r\n            </p>\r\n\n\n\n"
```
[the remark this answers is not in the document]
Comment (williamparry) [reply content/about/primary-page-content.json]: Can we have some more granular content control here?

[the remark this answers is not in the document]
Comment (BenRauzi) [reply content/about/primary-page-content.json]: @williamparry Yeah that's what I was saying about wanting to do some updates. I'll get onto them later.

[the remark this answers is not in the document]
Comment (BenRauzi) [reply content/about/primary-page-content.json]: The main issue with moving away from innerhtml is around how we do links in the body of paragraphs.

File: content/contributors/alex-chan.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "date": "2020-03-15T22:00:35.158Z",
```
Comment (williamparry) [inline content/contributors/alex-chan.json]: Adding arbitrary data doesn't seem like the right approach. These can be unordered.

**Comment by BenRauzi** [comment]
@williamparry Is XSS your main reason for wanting to move away from dangerouslySetInnerHTML or do you have others?

File: content/about/our-data.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
...
+  "order": 20
```
Comment (williamparry) [inline content/about/our-data.json]: Why is there an order property? The "about.json" page could be structured like "Andy bio", "Will bio", "What we do" etc all in the same content json. I don't think it's used elsewhere so no need to split it up.

Comment (williamparry) [reply content/about/our-data.json]: There's also <a href=\"#close\" title=\"Close\" class=\"modal-close\">\r\n                    Close\r\n                  </a> which should be handled by the page, not the content.

File: src/pages/about.js [hunk 4]
```diff
@@ -32,7 +36,7 @@ const AboutPage = ({data}) => {
                     __html: content_html,
                   }}>
                   </div>
-                  <br/><  br/>
+                  <br/><br/>
```
Comment (williamparry) [inline src/pages/about.js]: Can this be handled with CSS instead?

File: content/about/primary-page-content.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (williamparry) [reply content/about/primary-page-content.json]: For content links inside the body I can't find a solution other than using https://www.npmjs.com/package/dompurify to render it safe. Any non-content widget behaviour should be handled by the widget.

File: content/about/our-data.json (hunk 3 above)
Comment (BenRauzi) [reply content/about/our-data.json]: > Why is there an order property? The "about.json" page could be structured like "Andy bio", "Will bio", "What we do" etc all in the same content json. I don't think it's used elsewhere so no need to split it up.

The main reason I split these up was to make it easier-to-use an have more granular control over the content. It also took away some of the need to have raw html in the CMS and moves more content to be handled by the page. Happy to change it though.

File: content/contributors/alex-chan.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (williamparry) [inline content/contributors/alex-chan.json]: These could be an array on the about page

File: content/about/our-data.json (hunk 3 above)
Comment (williamparry) [reply content/about/our-data.json]: I don't see how the content structure being in multiple files would affect how the page would render it? There's little need to have something granular if it only exists in one place.

**Pull request merged**